qualtran/_infra/data_types.py

```diff
@@ -184,7 +184,7 @@ def num_qubits(self):
 
 
 @attrs.frozen
-class QFixedPoint(QDType):
+class QFxP(QDType):
```

Comment:
Sure, now means less pain later, but it's only a small amount of pain. Alternatively rather than a string we could just have sign, n_word and n_frac as attributes and could return a Fxp string for construction?

Reply:
I'd rather have a builder method using `n_word` and `n_frac` and keep the string as an attribute, especially if we plan to combine `QFxp` and `QUFxp`

Reply:
it's really easy to go from the actual attributes to a string and not really easy to go the other way...

Reply:
yeah I think 

``` python
qfxp = QFxp(signed=True, n_word=8, n_frac=7)
fxp = Fxp(float_val, dtype=qfxp.fxp_string())
```

is easier / clearer to type than

``` python
qfxp = QFxp(dtype="fxp-s8/7")
```

Reply:
Okay, storing and using explicit attributes SGTM. 

Just FYI - note that `n_word` for `Fxp` means total bitsize and in our previous convention, it used to mean bitsize for the integer part. We should pick a convention, document it and keep it consistent.

Reply:
right, I had to look at their docs before writing it. Currently we have int_bitsize and frac_bitsize and the signed bit is hiding there depending on if it's a QUFxp or a QFxp, but I think the n_word is more natural in some way. Given all this discussion I will pull out the Fxp renaming to a separate PR.